guide.md

```diff
@@ -0,0 +1,406 @@
+dotenv = "0.8.0"
+dotenv_macros = "0.8.0"
+```
+
+Diesel provides a separate [CLI][diesel-cli] tool to help manage your project.
+Since it's a standalone binary, and doesn't affect your project's code directly,
+we don't add it to `Cargo.toml`. Instead, we just install it on our system.
+
+[diesel-cli]: https://github.com/sgrif/diesel/tree/master/diesel_cli
+
+```shell
+cargo install diesel_cli
+```
+
+We need to tell Diesel where to find our database. We do this by setting the
+`DATABASE_URL` environment variable. On our development machines, we'll likely
+have multiple projects going, and we don't want to pollute our environment. We
+can put the url in a `.env` file instead.
+
+```shell
+echo DATABASE_URL=postgres://localhost/diesel_demo > .env
+```
+
+Now Diesel CLI can set everything up for us.
+
+```shell
+diesel setup
+```
+
+This will create our database for us (if it didn't already exist), and create an
+empty migrations directory that we can use to manage our schema (more on that
+later).
+
+Now we're going to write a small CLI that lets us manage a blog (ignoring the
+fact that we can only access the database from this CLI...). The first thing
+we're going to need is a table to store our posts. Let's create a migration for
+that:
+
+```shell
+diesel migration generate create_posts_table
+```
+
+You should see output which looks something like this:
+
+```
+Creating migrations/20160202154039_create_posts/up.sql
+Creating migrations/20160202154039_create_posts/down.sql
+```
+
+Migrations allow us to evolve the database schema over time. Each migration can
+be applied (`up.sql`) or reverted (`down.sql`). Applying and immediately
+reverting a migration should leave your database schema unchanged.
+
+```sql
+-- up.sql
+CREATE TABLE posts (
+  id SERIAL PRIMARY KEY,
+  title VARCHAR NOT NULL,
+  body TEXT NOT NULL,
+  published BOOLEAN NOT NULL DEFAULT 'f'
+)
+
+-- down.sql
+DROP TABLE posts
+```
+
+We can apply our new migration with `diesel migration run`. It's a good idea to
+make sure that `down.sql` is correct. You can do a quick sanity check by doing
+`diesel migration redo`, which will revert and reapply the latest migration.
+
+OK enough SQL, let's write some Rust. We'll start by writing some code to show
+us the last 5 published posts. The first thing we need to do is establish a
+database connection.
+
+```rust
+// src/lib.rs
+#[macro_use]
+extern crate diesel;
+extern crate dotenv;
+
+use diesel::prelude::*;
+use diesel::pg::PgConnection;
+use dotenv::dotenv;
+use std::env;
+
+pub fn establish_connection() -> PgConnection {
+    dotenv().ok();
+
+    let database_url = env::var("DATABASE_URL")
+        .expect("DATABASE_URL must be set");
+    PgConnection::establish(&database_url)
+        .expect(&format!("Error connecting to {}", database_url))
+}
+```
+
+We'll also want to create a `Post` struct that we can read our data into, and
```

Comment:
The dangling preposition on this clause makes me unhappy, can this read "into which we can read our data,"?